### ISSUE TITLE
0% trainable parameters when trying to use Config YAML - Phi4-8bit

### ISSUE THREAD

**Issue by JoshShailes** [ml-explore/mlx-lm#90]
Hi, I'm getting weird behaviour where I get 0% training parameters when trying to train Phi-4 models. Tested and works fine with mlx-community/Llama-3.2-1B-Instruct-4bit. If I pass the args via the command line then Phi 4 models will train. But I wanted to change the Rank and dropout values. Thanks.

model_path = "mlx-community/Phi-4-mini-instruct-8bit"

shell command: mlx_lm.lora --model mlx-community/Phi-4-mini-instruct-8bit --optimizer adamw --data data/ --train --config lora_config_default copy 4.yaml

```
Loading configuration file lora_config_default.yaml
Loading pretrained model
Loading datasets
Training
Trainable parameters: 0.000% (0.000M/3836.022M)
Starting training..., iters: 1000
```

Config File
```
# The path to the local model directory or Hugging Face repo.
model: "mlx_model"

# Whether or not to train (boolean)
train: true

# The fine-tuning method: "lora", "dora", or "full".
fine_tune_type: lora

# The Optimizer with its possible inputs
optimizer: adamw
optimizer_config:
   adamw:
     betas: [0.9, 0.98]
     eps: 1e-6
     weight_decay: 0.1
     bias_correction: true

# Directory with {train, valid, test}.jsonl files
data: "data/"

# The PRNG seed
seed: 0

# Number of layers to fine-tune
num_layers: 8

# Minibatch size.
batch_size: 4

# Iterations to train for.
iters: 1000

# Number of validation batches, -1 uses the entire validation set.
val_batches: 25

# Adam learning rate.
learning_rate: 1e-5

# Number of training steps between loss reporting.
steps_per_report: 10

# Number of training steps between validations.
steps_per_eval: 20

# Load path to resume training with the given adapter weights.
resume_adapter_file: null

# Save/load path for the trained adapter weights.
adapter_path: "adapters"

# Save the model every N iterations.
save_every: 100

# Evaluate on the test set after training
test: false

# Number of test set batches, -1 uses the entire test set.
test_batches: 100

# Maximum sequence length.
max_seq_length: 2048

# Use gradient checkpointing to reduce memory use.
grad_checkpoint: false

# LoRA parameters can only be specified in a config file
lora_parameters:
  # The layer keys to apply LoRA to.
  # These will be applied for the last lora_layers
  keys: ["self_attn.q_proj", "self_attn.v_proj"]
  rank: 4
  scale: 20.0
  dropout: 0.1

# Schedule can only be specified in a config file, uncomment to use.
#lr_schedule:
#  name: cosine_decay
#  warmup: 100 # 0 for no warmup
#  warmup_init: 1e-7 # 0 if not specified
#  arguments: [1e-5, 1000, 1e-7] # passed to scheduler

#hf_dataset:
#  path: "billsum"
#  train_split: "train[:1000]"
#  valid_split: "train[-100:]"
#  prompt_feature: "text"
#  completion_feature: "summary"
```

**Comment by JoshShailes** [comment]
lora_parameters:
  # The layer keys to apply LoRA to.
  # These will be applied for the last lora_layers
  **keys: ["self_attn.q_proj", "self_attn.v_proj"]**
  rank: 4
  scale: 20.0
  dropout: 0.1


So I "think" the issue is with the keys here but not 100% sure what I should change them to?

**Comment by JoshShailes** [comment]
Closing as Solved:

Changed to "self_attn.qkv_proj"
